spec/helpers/react_helper_spec.rb

```diff
@@ -124,5 +124,20 @@
           with('SearchResultsLayout', hash_including(relatedSearches: [related_search]))
       end
     end
+
+    context 'when affiliate has a non-blank external_tracking_code' do
+      let(:external_tracking_code) { '<script>var foo;</script>' }
+
+      before do
+        allow(affiliate).to receive(:external_tracking_code).and_return(external_tracking_code)
```

Comment:
There is not need to use a mock in here. This can be done with a regular assignment